app/code/Magento/InventoryInStorePickupShippingApi/Model/Carrier/Validation/RequestValidatorChain.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickupShippingApi\Model\Carrier\Validation;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Validation\ValidationResult;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\InventoryInStorePickupShippingApi\Model\Carrier\Validation\RequestValidatorInterface;
+use Magento\Quote\Model\Quote\Address\RateRequest;
+
+/**
+ * @inheritdoc
+ */
+class RequestValidatorChain implements RequestValidatorInterface
+{
+    /**
+     * @var ValidationResultFactory
+     */
+    private $validationResultFactory;
+
+    /**
+     * @var RequestValidatorInterface[]
+     */
+    private $validators;
+
+    /**
+     * @param ValidationResultFactory $validationResultFactory
+     * @param array $validators
+     *
+     * @throws LocalizedException
+     */
+    public function __construct(
+        ValidationResultFactory $validationResultFactory,
+        array $validators = []
+    ) {
+        $this->validationResultFactory = $validationResultFactory;
+
+        foreach ($validators as $validator) {
+            if (!$validator instanceof RequestValidatorInterface) {
+                throw new LocalizedException(
```

Comment:
I'd change it to `InvalidArgumentException`, this kind of error does not make sense to customers anyway.

Reply:
Done.